### ISSUE TITLE
Duplicate deposits to ScholarSphere

### ISSUE THREAD

**Issue by anaelizabethenriquez** [psu-libraries/researcher-metadata#755]
Paulina has reported that ScholarSphere is getting duplicate deposits of articles from RMD. The example I have right now is RMD publication 533969, which has the following two deposits:

- https://scholarsphere.psu.edu/resources/67450b81-6bb1-4d00-befb-6dd73a3015c2
- https://scholarsphere.psu.edu/resources/cd1410e4-9bac-44c5-8a3b-5e163d06b451

That publication has only one import associated with it, so I don't think it's the product of a duplicate merge. It's not in a duplicate group now. And I don't see any other records in RMD that look like they ought to be merged with it.

Both ScholarSphere records are linked from this publication. I don't understand how the user was able to deposit the article a second time, since it was already in ScholarSphere.

I don't have any other examples right now, but Paulina said there are several of these. I'll update this issue if I learn about more.

**Comment by ajkiessl** [comment]
Okay, I figured out the rest of it.  Judging by the ScholarSphere ingest endpoint code, we aren't stopping the creation of a record if there's a PsuIdentity::SearchService::NotFound error like we encountered with the first example above.  I can add an issue to the ScholarSphere backlog to handle this differently.  We should either be returning a 200, creating a record, and then returning some kind of warning message, _or_ returning a 400 and not create the record.

For last example above, those are in fact work:work_version pairs.  These must've gotten around our attempts to clean these out of our OpenAccessLocations.  Not too sure how, though.

**Comment by anaelizabethenriquez** [comment]
Thanks @ajkiessl! Sounds like with the issue you added to the ScholarSphere backlog, we should be covered on this. Looping in @PaulinaKrys so she can see all this. Paulina, if you see any more examples of this, please let me know.

**Comment by PaulinaKrys** [comment]
Thanks @ajkiessl and @anaelizabethenriquez! I actually went through some more deposits and found a couple of additional example pairs for you to take a look at. All cases appear to be more similar to the last example from earlier where the metadata varied between submissions (formatting of the author names, etc.):

- https://scholarsphere.psu.edu/resources/67914f4d-88cd-4a54-9f8a-6856075507a2 
- https://scholarsphere.psu.edu/resources/072d850d-8e56-414f-a090-2df98c7a48f1

- https://scholarsphere.psu.edu/resources/c1ff2067-750c-438d-95e8-b7648d0c07f1
- https://scholarsphere.psu.edu/resources/34b26e26-9574-4eb1-8a41-492df25e90bb

- https://scholarsphere.psu.edu/resources/4636d229-a0e9-4e9b-a39b-67ef59a785c0
- https://scholarsphere.psu.edu/resources/90da92e8-77da-4df1-8404-c61e35f1519c

- https://scholarsphere.psu.edu/resources/6e94ca1c-0b14-40f0-85d7-1b285ad7ee57
- https://scholarsphere.psu.edu/resources/a323679c-805b-4cb1-9ca8-6fcc6053b773

**Comment by ajkiessl** [comment]
@PaulinaKrys Correct, these are all a product of duplicates in RMD that were not grouped at some point.  2 and 4 have since been grouped and merged.  1 and 3 have not yet been grouped.  I'm not too sure why the ones that are still not grouped are this way.  They seem to fit the criteria: similar titles, publication dates, same DOI.  I'll have to investigate this further.

**Comment by avshoff** [comment]
@anaelizabethenriquez I will be adding a few duplicate deposits in ScholarSphere from RMD. I'll drop the links, but please let me know if there's any other information I can provide that would be helpful. 

- https://scholarsphere.psu.edu/resources/81752ae5-68a6-4284-b713-4cdb077794cc
- https://scholarsphere.psu.edu/resources/6a268b2a-788b-4bef-9408-f3a167fc3ca1

**Comment by avshoff** [comment]
Lagoa - Probabilistic Discrete Time Robust H2 Controller Design

- https://scholarsphere.psu.edu/resources/baefdfcd-6171-4dd3-9cc4-af814908dd44
- https://scholarsphere.psu.edu/resources/8f38af0f-cfcf-49d6-8c94-c3b3b466d0c3

**Comment by anaelizabethenriquez** [comment]
Related issues #504 #386

**Comment by avshoff** [comment]
McCrudden - Effects of emotions, topic beliefs, and task instructions on the processing and recall of a dual-position text

- https://scholarsphere.psu.edu/resources/af4c6ecd-9891-44f5-980c-f3195228a4dd
- https://scholarsphere.psu.edu/resources/9f38bf47-9b24-46cc-b586-1bdc57de54b2

**Comment by avshoff** [comment]
Bakis - Effect of Moisture on the Tensile Properties of Composites with Bio-based Fibers and Matrix

- https://scholarsphere.psu.edu/resources/fdb9243e-b92e-4158-970e-d1958e036658
- https://scholarsphere.psu.edu/resources/8eea0b30-b968-4f44-b7d4-125d8f736a75

**Comment by avshoff** [comment]
Rosenbaum - Coping with childhood maltreatment: Avoidance and eating disorder symptoms (deposited 8/1/22)

- https://scholarsphere.psu.edu/resources/a323679c-805b-4cb1-9ca8-6fcc6053b773
- https://scholarsphere.psu.edu/resources/6e94ca1c-0b14-40f0-85d7-1b285ad7ee57

**Comment by avshoff** [comment]
Albert - synergy: A Python library for calculating, analyzing, and visualizing drug combination synergy (deposited 8/1/22)

- https://scholarsphere.psu.edu/resources/f130a18b-0332-4bba-b15f-a577458a6c97
- https://scholarsphere.psu.edu/resources/dec56da3-7c5a-431d-80af-cd0ec206aa74

**Comment by avshoff** [comment]
Price: Fine-structure-resolved rovibrational transitions for SO+H2 collisions/Fine-structure resolved rovibrational transitions for so + H<sub>2</sub>collisions (deposited July 25, 2022)

- https://scholarsphere.psu.edu/resources/c1ff2067-750c-438d-95e8-b7648d0c07f1
- https://scholarsphere.psu.edu/resources/34b26e26-9574-4eb1-8a41-492df25e90bb

**Comment by avshoff** [comment]
Muir: Initial development of the Dance Imagery Questionnaire for Children (DIQ-C): establishing content validity (deposited December 6, 2023)

- https://scholarsphere.psu.edu/resources/3908cd87-ec3d-4357-ab72-b8cfd57759eb
- https://scholarsphere.psu.edu/resources/fd4f461d-a9b1-4af9-b0e1-d1fa2638421b

**Comment by avshoff** [comment]
Case: Curved Versions of the Ovsienko–Redou Operators (deposited January 23, 2024)

- https://scholarsphere.psu.edu/resources/794e61d4-0154-4066-b6a9-3f5ed35f8e8e
- https://scholarsphere.psu.edu/resources/b3c03984-afde-4be3-81a0-763eb4428c4c

**Comment by anaelizabethenriquez** [comment]
@ajkiessl Want to make sure you see @avshoff 's comment above about Case: Curved Versions of the Ovsienko–Redou Operators (deposited January 23, 2024).

This is a weird one, because both deposits were made with the AI OA Workflow, using the same file, on the same publication record. I didn't think that was possible.

**Comment by ajkiessl** [comment]
@anaelizabethenriquez My assumption here is that when depositing, an admin opened the deposit page twice for the same record in separate tabs and then submitted each one.  While I'm not 100% certain, this is the only thing I could think of that could lead to this.  I was also able to recreate this behavior on QA.  Ideally there should be something in the code to stop this from happening, but there currently is not.  I think that should be an easy fix, though, so I can look into patching it ASAP.

**Comment by EricDurante** [comment]
Since I'm here lurking...

What about a double-click on the submit button? Could that have caused it also? Either way, the solution is probably the same.

**Comment by ajkiessl** [comment]
Ah thanks Eric.  Yes, that must be it and more likely than my scenario.  I tested that out in QA and it created duplicates.  I should have a PR ready for this fix soon.

**Comment by ajkiessl** [comment]
@anaelizabethenriquez The double-clicking issue has been resolved and deployed to production

**Comment by avshoff** [comment]
First set of duplicate deposits via RMD in a while: 
- https://scholarsphere.psu.edu/resources/5923a866-8774-4100-8131-bee1d98eebf3
- https://scholarsphere.psu.edu/resources/29ffa48a-feb4-4761-ae03-d4a734685eda
- https://scholarsphere.psu.edu/resources/49593d3f-f7a8-4470-8d91-a8eef2f9d6ba
